### ISSUE TITLE
[TW-1510] Task can save empty value in the data backlog

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1536]
_Tomas Babej on 2015-01-13T18:57:06Z says:_ 

 It is sufficient to run:

```
$ task add test "pro:''"
```

or

```
$ task add test "pri:''"
```

And you will end up getting
```
cat ~/.task/backlog.data 
{"description":"test","entry":"20150113T185847Z","modified":"20150113T185847Z","project":"","status":"pending","uuid":"aa6edc65-3a25-48a0-a920-6c54dc8af981"}
{"description":"test","entry":"20150113T185918Z","modified":"20150113T185918Z","priority":"","status":"pending","uuid":"fa213e41-b5ec-4e7f-a6d0-768f8f37bb42"}
```

In the backlog.data, not in other .data files though.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-01-13T18:57:06Z
Modified: 2015-02-14T21:40:34Z
```

**Comment by taskwarrior** [comment]
_Tomas Babej on 2015-02-14T19:00:23Z says:_ 

 Actually, the following example is enough to trigger the bug:

{code}
$ task add test pro:
{code}

**Comment by taskwarrior** [comment]
_Tomas Babej on 2015-02-14T21:40:34Z says:_ 

 Fixed with: 328b7ec2604c66ed758692fe8e1577c96b56f7f7

**Comment by pbeckingham** [comment]
[TW-1510_tbabej-task-0017-Task-Do-not-export-empty-attributes-in-composeJSON.patch.txt](https://github.com/GothenburgBitFactory/taskwarrior/files/1756377/TW-1510_tbabej-task-0017-Task-Do-not-export-empty-attributes-in-composeJSON.patch.txt)
